kaldo/forceconstants.py

```diff
@@ -235,3 +211,45 @@ def unfold_third_order(self, reduced_third=None, distance_threshold=None):
         expanded_third = expanded_third.reshape(
             (n_unit_atoms * 3, n_replicas * n_unit_atoms * 3, n_replicas * n_unit_atoms * 3))
         return expanded_third
+
+
+    def elastic_prop(self):
+
+        # Intake key parameters
+        atoms = self.atoms
+        masses = atoms.get_masses()
+        volume = atoms.get_volume()
+        list_of_replicas = self.second.list_of_replicas
+        h0 = HarmonicWithQ(np.array([0, 0, 0]), self.second, storage='numpy')
+        dynmat = self.second.dynmat[0]  # units THz^2
+        positions = self.atoms.positions
+        n_unit = atoms.positions.shape[0]
+        e_mu = np.array(h0._eigensystem[1:, :]).reshape(
+            (n_unit, 3, 3 * (n_unit)))
+        w_mu = np.abs(np.array(h0._eigensystem[0, :])) ** (0.5)  # optical frequencies (w/(2*pi) = f) in THz
+        distance = positions[:, np.newaxis, np.newaxis, :] - (
+                    positions[np.newaxis, np.newaxis, :, :] + list_of_replicas[np.newaxis, :, np.newaxis, :])
+        d1 = np.einsum('iljx,ibljc->ibjcx',
+                       tf.convert_to_tensor(distance.astype(complex)),
+                       tf.cast(dynmat, tf.complex128))  # THz^2*Ang (it should be multiplied by i?)
+        d2 = -1 * np.einsum('iljx,iljy,ibljc->ibjcxy',
+                            tf.convert_to_tensor(distance.astype(complex)),
+                            tf.convert_to_tensor(distance.astype(complex)),
+                            tf.cast(dynmat, tf.complex128))  # THz^2*Ang^2
+        gamma = np.einsum('iav,jbv,v->iajb', e_mu[:, :, 3:], e_mu[:, :, 3:], 1 / w_mu[3:] ** 2)  # Gamma tensor from paper
+        
+        # Compute component b and r, keep the real component only
+        b = (1/(2*volume))*np.einsum('n,m,nimjkl->ijkl', masses**(0.5), masses**(0.5), d2).real
+        d1r = np.einsum('nhmij,m->nhmij', d1, masses**(0.5))
+        r = -1 * (1/volume) * np.einsum('nhmij,nhrp,rpskl->ijkl', d1r, gamma, d1r).real
+        cijkl = np.zeros((3, 3, 3, 3))
+        evtotenjovermol = units.mol / (10 * units.J)
+        evperang3togpa = 160.21766208
```

Comment:
can we write this using ase?

Reply:
You mean the unit conversion factor and units correct? Sure, I can redefine the unit conversion factor with ase or scipy.units

Reply:
`ase.units` would be ideal to keep it consistent with the rest of the code.